core/src/apps/common/address_type.py

```diff
@@ -24,7 +26,7 @@ def check(address_type: int, raw_address: bytes) -> bool:
 
 def strip(address_type: int, raw_address: bytes) -> bytes:
     if not check(address_type, raw_address):
-        raise ValueError("Invalid address")
+        raise wire.DataError("Invalid address")
```

Comment:
i'd argue that this is also a bad place for a wire error, and the caller should make sure to validate the address (and return a wire error) before calling these functions